ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/MachineTestsTable.test.tsx

```diff
@@ -275,4 +288,134 @@ describe("MachineTestsTable", () => {
       wrapper.find("Button[data-test='action-menu-show-metrics']").exists()
     ).toEqual(false);
   });
+
+  it("sends an analytics event when suppressing a script result", () => {
+    state.nodescriptresult.items = { abc123: [1] };
+    const scriptResults = [
+      scriptResultFactory({
+        id: 1,
+        result_type: ResultType.Testing,
+        status: scriptStatus.FAILED,
+        suppressed: false,
+      }),
+    ];
+    state.scriptresult.items = scriptResults;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineTestsTable machineId="abc123" scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    const checkbox = wrapper.find('input[data-test="suppress-script-results"]');
+    checkbox.simulate("change", { target: { value: "checked" } });
+
+    expect(mockSendAnalytics).toHaveBeenCalled();
+    expect(mockSendAnalytics.mock.calls[0]).toEqual([
+      "Machine testing",
+      "Suppress script result failure",
+      "Suppress",
+    ]);
+  });
+
+  it("sends an analytics event when unsuppressing a script result", () => {
+    state.nodescriptresult.items = { abc123: [1] };
+    const scriptResults = [
+      scriptResultFactory({
+        id: 1,
+        result_type: ResultType.Testing,
+        status: scriptStatus.FAILED,
+        suppressed: true,
+      }),
+    ];
+    state.scriptresult.items = scriptResults;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineTestsTable machineId="abc123" scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    const checkbox = wrapper.find('input[data-test="suppress-script-results"]');
+    checkbox.simulate("change", { target: { value: "checked" } });
```

Comment:
Shouldn't this simulate unchecking the check box?

Reply:
Yeah it should actually. Funny that it works at all - looks like it's because the thing that determines whether `suppress` or `unsuppress` is called is based on the test and not on the checkbox state.